### PR TITLE
Issue pg-irc/pathways-backend#251 Add missing schema validation tests…

### PR DESCRIPTION
**Pull request by bensinjin**
… for address fields, organize tests into logical groups, allow null values for address, state_province, and postal_code

### CODE REVIEW

**Comment by bensinjin** [comment]
This fixes our crash on tasks that pull in services with null values for their postal code.

**Comment by bensinjin** [comment]
I likely should have put the reorganization of the schema tests into its own commit. I'll be sure to remember to better isolate noisy refactoring commits.

File: src/stores/__tests__/helpers/services_schema_helpers.ts [hunk 1]
```diff
@@ -71,34 +71,39 @@ export class PhoneNumberJSONBuilder {
...
-    id: number = anInteger();
-    address: string | null = aString();
-    city: string | null = aString();
-    state_province: string | null = aString();
-    postal_code: string | null = aString();
-    country: string | null = aString();
-
-    withAddress(address: string | null): AddressJSONBuilder {
+    id: any = anInteger();
+    address: any | number = aString();
```
Comment (rasmus-storjohann-PG) [inline src/stores/__tests__/helpers/services_schema_helpers.ts]: `any | number` type is equivalent to just `any`

File: src/stores/__tests__/helpers/services_schema_helpers.ts [hunk 2]
```diff
@@ -1,29 +1,29 @@
-// tslint:disable:no-class readonly-keyword no-expression-statement no-this no-let
+// tslint:disable:no-class readonly-keyword no-expression-statement no-this no-let no-any
 import { aString, anInteger } from '../../../application/__tests__/helpers/random_test_values';
 
 interface PhoneNumberJSON {
-    readonly phone_number_type?: string | null;
-    readonly phone_number?: string | null;
+    readonly phone_number_type?: any;
+    readonly phone_number?: any;
```
Comment (rasmus-storjohann-PG) [inline src/stores/__tests__/helpers/services_schema_helpers.ts]: So why are you changing `string | null` to `any`?

File: src/stores/__tests__/helpers/services_schema_helpers.ts (hunk 1 above)
Comment (bensinjin) [reply src/stores/__tests__/helpers/services_schema_helpers.ts]: Find and replace error. Fixed.

File: src/stores/__tests__/helpers/services_schema_helpers.ts (hunk 2 above)
Comment (bensinjin) [reply src/stores/__tests__/helpers/services_schema_helpers.ts]: `null` used to be used to represent the invalid type. Now `null` is valid in some cases so we'd need to pick a new invalid type. Naturally we could tack on ` | number` but I felt that would confuse things. These types exist for the soul purpose of letting us test the schemas with missing values and values of incorrect types maybe the naming could be improved to better reflect this?

Comment (rasmus-storjohann-PG) [reply src/stores/__tests__/helpers/services_schema_helpers.ts]: OK, the reason is good but the situation is unusual. I would add a comment explaining this. I would probably break out the `no-any` from 
```ts
// tslint:disable:no-class readonly-keyword no-expression-statement no-this no-let no-any
```
to have it on its own with the comment explaining, but there are other ways to do it too.

Comment (bensinjin) [reply src/stores/__tests__/helpers/services_schema_helpers.ts]: I've added a comment.

**Pull request merged**